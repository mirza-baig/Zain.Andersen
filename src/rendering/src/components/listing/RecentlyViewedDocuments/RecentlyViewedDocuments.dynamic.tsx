// Global
import { useState } from 'react';
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import { Component } from 'src/helpers/Component';
import { Headline } from 'src/helpers/Headline';
import { SvgIcon } from 'src/helpers/SvgIcon';
import { LinkWrapper } from 'src/helpers/LinkWrapper';
import { RecentlyViewedDocumentsTheme } from './RecentlyViewedDocuments.theme';
import classNames from 'classnames';

export type RecentlyViewedDocumentsProps =
  Feature.EnterpriseWeb.AndersenWindows.Components.Listing.RecentlyViewedDocuments.RecentlyViewedDocuments;
const RecentlyViewedDocuments = (props: RecentlyViewedDocumentsProps) => {
  const { themeData, themeName } = useTheme(RecentlyViewedDocumentsTheme);
  const [showAll, setShowAll] = useState(false);
  const recentlyViewedItems = props.recentlyViewedLinks?.recentLinks || [];

  return (
    <Component variant="lg" dataComponent="listing/recentlyvieweddocuments" {...props}>
      {recentlyViewedItems.length > 0 && (
        <>
          <div className={themeData.classes.headlineBorder}>
            <div className="mt-s flex md:mt-xxs">
              {themeName === 'aw' && (
                <div className="m-xxxs mt-[1px] flex">
                  <SvgIcon icon="orange-triangle" />
                </div>
              )}
              <Headline classes={themeData.classes.headlineClass} {...props} />
            </div>
          </div>

          <div className="col-span-12 border-t border-solid border-gray md:col-span-8">
            <ul className={`flex flex-col`}>
              {recentlyViewedItems.length > 0 &&
                recentlyViewedItems.map((item: RecentlyViewedDocumentsProps, index: number) => {
                  return (
                    item && (
                      <li key={index} className={classNames(!showAll && index > 3 && 'hidden')}>
                        <LinkWrapper
                          field={item}
                          className="flex border-b border-solid border-gray py-s text-body text-darkprimary hover:underline"
                          target="_blank"
                          ariaLabel={{
                            value: item || 'Recently Viewed Items',
                          }}
                        />
                      </li>
                    )
                  );
                })}
            </ul>
            {recentlyViewedItems.length > 4 && (
              <div
                className="mt-s cursor-pointer font-sans text-text-link font-heavy underline"
                onClick={() => setShowAll(!showAll)}
              >
                {showAll ? 'View Less' : 'View All'}
              </div>
            )}
          </div>
        </>
      )}
    </Component>
  );
};

export default withDatasourceCheck()<RecentlyViewedDocumentsProps>(RecentlyViewedDocuments);
