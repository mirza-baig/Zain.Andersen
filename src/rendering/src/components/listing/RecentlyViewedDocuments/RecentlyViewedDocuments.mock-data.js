const defaultData = {
  rendering: {
    dataSource: 'RecentlyViewedDocuments',
    componentName: 'RecentlyViewedDocuments',
  },

  fields: {
    componentSpacing: null,
    sectionId: {
      value: '',
    },
    headlineLevel: {
      id: '0f556f3a-bbad-4aa5-952d-b79003b39cd6',
      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2',
      name: 'Heading 2',
      displayName: 'Heading 2',
      fields: {
        Value: {
          value: 'h2',
        },
      },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    headlineText: {
      value: 'Recently Viewed',
    },
    eventName: {
      value: '',
    },
    eventType: {
      value: '',
    },
    eventZone: {
      value: '',
    },
  },
};

export default defaultData;
