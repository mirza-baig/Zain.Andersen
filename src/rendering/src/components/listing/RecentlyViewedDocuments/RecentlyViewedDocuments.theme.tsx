// Lib
import { ThemeFile } from 'lib/context/ThemeContext';

export const RecentlyViewedDocumentsTheme: ThemeFile = {
  aw: {
    classes: {
      headlineBorder: 'col-span-12 border-t-2 border-black md:col-span-3',
      headlineClass: 'text-text-link lg:text-xs font-heavy uppercase mr-1',
    },
  },
  rba: {
    classes: {},
  },
};
