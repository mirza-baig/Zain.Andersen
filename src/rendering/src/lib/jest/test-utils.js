// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
