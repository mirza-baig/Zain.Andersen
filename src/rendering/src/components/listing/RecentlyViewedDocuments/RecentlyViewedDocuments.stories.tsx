// Global
import { Story, Meta } from '@storybook/react';
// Local
import RecentlyViewedDocuments, {
  RecentlyViewedDocumentsProps,
} from './RecentlyViewedDocuments.dynamic';
import defaultData from './RecentlyViewedDocuments.mock-data';

export default {
  title: 'Listing/RecentlyViewedDocuments',
  component: RecentlyViewedDocuments,
} as Meta;
const Template: Story<RecentlyViewedDocumentsProps> = (props) => (
  <RecentlyViewedDocuments {...props} />
);
export const Default = Template.bind({});
Default.args = defaultData;
