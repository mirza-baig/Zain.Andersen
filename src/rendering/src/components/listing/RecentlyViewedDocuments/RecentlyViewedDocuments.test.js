// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import RecentlyViewedDocuments from './RecentlyViewedDocuments.dynamic';
import defaultData from './RecentlyViewedDocuments.mock-data';

it('renders correctly', async () => {
  const component = snapshot(RecentlyViewedDocuments, { componentProps: defaultData });
  await hasDataComponent(component, 'listing/recentlyvieweddocuments');
});
