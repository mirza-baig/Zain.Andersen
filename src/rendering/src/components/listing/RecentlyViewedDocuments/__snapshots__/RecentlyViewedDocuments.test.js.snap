// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="listing/recentlyvieweddocuments"
  id="id1960888660"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    />
  </div>
</section>
`;
